### PR TITLE
Library to perform 3 way merge algorithm

### PR DESCRIPTION
**Pull request by rakesh-verma-16**
Initially making it to work with simple arrays when there will be no conflict. After that, we will start add complex features to handle conflicts.

### CODE REVIEW

**Comment by dickolsson** [comment]
@sharprakeshverma The basic concept here looks good. Before I merge this I'd like you to clean up the code to strictly confirm to PSR-2 code styling.

File: src/ThreeWayMerge.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+             * Else If ancestor's value is equal to local's value
+             * Set the merged array's value to remote value.
+             *
+             * Else If local's value is equal to remote's value
+             * Set the merged array's value to any value
+             * as they both are same.
+             *
+             * Else set conflict to TRUE as none of the above is True.
+             *
+             */
```
Comment (dickolsson) [inline src/ThreeWayMerge.php]: The comments above should just be inline comments, using `//` and not `/**`.

File: tests/ThreeWayMergeTest.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Relaxed\Merge\Test;
+
+require __DIR__ . '/../vendor/autoload.php';
```
Comment (dickolsson) [inline tests/ThreeWayMergeTest.php]: You should be able to remove this now when autoloading seems to be setup correctly.

File: tests/ThreeWayMergeTest.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        $remote = [
+            'title' => '123',
+            'body' => 'lorem ipsum',
+        ];
+
+        $merge = new ThreeWayMerge();
+        $result = $merge->performMerge($original, $local, $remote);
+        $expected = [
+            'title' => '123',
+            'body' => "dolor",
```
Comment (dickolsson) [inline tests/ThreeWayMergeTest.php]: Use single quotation marks `'` for consistency

File: tests/ThreeWayMergeTest.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    /**
+     * Test function for cases when merge conflict would arise.
+     * @throws Exception
+     */
+    public function testConflict()
+    {
+        $original = [
+            'title' => 'abc',
+            'body' => "lorem ipsum",
```
Comment (dickolsson) [inline tests/ThreeWayMergeTest.php]: Use single quotation marks `'` for consistency

File: src/ThreeWayMerge.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        // Returning a new Array for now. Can return the modified ancestor as well.
+        $merged = $ancestor;
+        $conflict = false;
+        foreach ($ancestor as $key => $value) {
+            // If ancestor's value is equal to remote's value
+            // Set the merged array's value to local value
+
+            // Else If ancestor's value is equal to local's value
+            //Set the merged array's value to remote value.
```
Comment (dickolsson) [inline src/ThreeWayMerge.php]: Space needed after `//`

File: src/ThreeWayMerge.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+            // Else If local's value is equal to remote's value
+            //Set the merged array's value to any value
+            //as they both are same.
```
Comment (dickolsson) [inline src/ThreeWayMerge.php]: Space needed after `//`

File: src/ThreeWayMerge.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+            // Else set conflict to TRUE as none of the above is True/
```
Comment (dickolsson) [inline src/ThreeWayMerge.php]: End line with a dot instead of `/`.

File: src/ThreeWayMerge.php [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        $merged = $ancestor;
+        $conflict = false;
+        foreach ($ancestor as $key => $value) {
+            // If ancestor's value is equal to remote's value
+            // Set the merged array's value to local value
+
+            // Else If ancestor's value is equal to local's value
+            //Set the merged array's value to remote value.
+
+            // Else If local's value is equal to remote's value
```
Comment (dickolsson) [inline src/ThreeWayMerge.php]: Sentence should end with full stop, i.e. a dot.

File: src/ThreeWayMerge.php [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    public function performMerge(array $ancestor, array $local, array $remote)
+    {
+        // Returning a new Array for now. Can return the modified ancestor as well.
+        $merged = $ancestor;
+        $conflict = false;
+        foreach ($ancestor as $key => $value) {
+            // If ancestor's value is equal to remote's value
+            // Set the merged array's value to local value
+
+            // Else If ancestor's value is equal to local's value
```
Comment (dickolsson) [inline src/ThreeWayMerge.php]: Sentences above should end with full stop, i.e. a dot.

File: composer.json [hunk 10]
```diff
@@ -1,4 +1,12 @@
 {
```
Comment (jeqq) [inline composer.json:1]: @sharprakeshverma You should also add the autoload for test and phpunit as dependence.

File: tests/ThreeWayMergeTest.php [hunk 11]
```diff
@@ -0,0 +1,70 @@
+<?php
```
Comment (jeqq) [inline tests/ThreeWayMergeTest.php]: @sharprakeshverma Add `phpunit.xml.dist` configuration file.

File: phpunit.xml.dist [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+         colors="true"
+         bootstrap="vendor/autoload.php"
+>
+
+    <testsuites>
+        <testsuite name="Merge Library Test Suite">
+            <directory>tests</directory>
+        </testsuite>
+    </testsuites>
+</phpunit>
```
Comment (jeqq) [inline phpunit.xml.dist]: Here should be a new empty line.

**Comment by dickolsson** [comment]
This PR looks good now. If there's any cleaning up to still do we can do that in a follow-up PR :)

**Pull request merged**